server/middleware/mentor/index.js

```diff
@@ -12,41 +13,164 @@ const resObj = (statusCode, message, data = {}) => ({
+    ...(mentor_id ? { mentor_id: mentor_id } : {}),
+    ...(bio ? { bio: bio } : {}),
+    ...(skills ? { skills: skills } : {}),
+    ...(phone ? { phone: phone } : {}),
+    ...(timezone ? { timezone: timezone } : {}),
+    ...(googlemeet ? { googlemeet: googlemeet } : {}),
+    ...(timeSlot ? { timeSlot: timeSlot } : {}),
+  };
+};
+
+const saveMentorInfo = async (req, res, tokenData) => {
   let data;
   let statusCode;
   let message;
-  let collectionObj;
+  const { USERS_COLLECTION } = process.env;
+  const { MENTORS_COLLECTION } = process.env;
 
-  logger.info(
-    setLogDetails(
-      'mentorMiddleware.getMentors',
-      'Fetching metors data',
-      `User - ${email}`
-    )
-  );
+  try {
+    const mentorPayload = createPayload(req.body, tokenData.user_id);
+    const mentorCollection = await mongoUtil.fetchCollection(
+      MENTORS_COLLECTION
+    );
+    const saveInfo = await mentorCollection.insertOne(mentorPayload);
+    const { password, ...rest } = saveInfo.ops[0];
+    data = rest;
+    if (!data) {
+      throw Error('Something went wrong!');
+    } else {
+      statusCode = OK;
+      message = 'Success';
+    }
+    const userCollection = await mongoUtil.fetchCollection(USERS_COLLECTION);
+    const updateUser = await userCollection.findOneAndUpdate(
+      { email: tokenData.email },
+      { $set: { hasOnboarded: true } },
+      { returnOriginal: false }
+    );
+    const { pass, ...value } = updateUser.value;
+    generateCookie(res, value);
+  } catch (err) {
+    if (err) {
+      statusCode = statusCode || BAD_REQUEST;
+      message = err.message;
+    } else {
+      statusCode = GATEWAY_TIMEOUT;
+      message = 'Gateway timeout';
+    }
+  }
+  console.log(statusCode, message, data);
+  return { statusCode, message, data };
+};
 
+const updateMentorInfo = async (req, res, tokenData) => {
+  let data;
+  let statusCode;
+  let message;
+  try {
+    const mentorPayload = createPayload(req.body, tokenData.user_id);
+    const { MENTORS_COLLECTION } = process.env;
+    const mentorCollection = await mongoUtil.fetchCollection(
+      MENTORS_COLLECTION
+    );
+    if (req.body.hasOnboarded) {
+      const result = await saveMentorInfo(req, res, tokenData);
+      data = result.data;
+      statusCode = result.statusCode;
+      message = result.message;
+    } else {
+      const updateProfile = await mentorCollection.findOneAndUpdate(
+        { mentor_id: tokenData.user_id },
+        { $set: mentorPayload },
+        { returnOriginal: false }
+      );
+      if (!updateProfile.value) {
+        statusCode = 401;
+        throw Error('User does not exist!');
+      }
+      const { password, ...rest } = updateProfile.value;
+      statusCode = OK;
+      message = 'Success';
+      data = rest;
+    }
+  } catch (err) {
+    if (err) {
+      statusCode = statusCode || BAD_REQUEST;
+      message = err.message;
+    } else {
+      statusCode = GATEWAY_TIMEOUT;
+      message = 'Gateway timeout';
+    }
```

Comment:
log error events

Reply:
resolve in commit #https://github.com/femmecubator/femmecubator-api-repo/pull/72/commits/58b6dd83fe0bda2e2c2dc2899c8593f45bc7bcc0